cp23/hpi-fun2/maze/problem.tex

```diff
@@ -0,0 +1,65 @@
+\input{template.tex}
+
+\begin{document}
+
+\makeheader
+
+One night Alice wakes up in a dark room. She has read about this place in one of her books. It is a large maze that is known to have exactly one exit in it. But Alice is running out of time. If she doesn't find her way to the exit soon, she will be trapped in it forever.
+
+The only thing that Alice knows about this place is that it is build in a grid style. Each room in the grid is either a room or completely blocked. She can only travel to one of the four rooms directly adjacent to her current room.
+
+Since it is very dark, she can only feel in the room she is currently in, if the adjacent rooms are open or blocked. Can you help Alice find a way out of the maze before she will get trapped there forever?
+
+\paragraph*{Interaction}
+
+This is an interactive problem. This means that your program will not first read all intput and then write some output, but instead communicates with the jury by writing queries to standard output and reading the answers from standard input.
+
+In the first line, you recieve two integer values $w$ and $h$ ($3\leq w, h \leq 10^9$), the width and height of the maze.
```

Comment:
You should probably bound the product of w and h because as is the problem isn't always solvable.

Reply:
Actually, it's possible to solve the problem without being given w and h, so you might consider removing this if you want to. It's also fine to leave as is though.

Reply:
We also thought about this (leaving out w and h) but then we couldn't give a max number of interactions and to solve the problem one would need to use a map instead of a vector.

To you think this problem would be to easy if we provide `w` and `h`?

Reply:
I just changed to problem statement (and decided to bound it to `w * h <= 10^6`. Not sure if we should go even higher since runtime is `O(4* w* h)`.

Reply:
I wouldn't go any higher, actually you probably need to lower it a bit further because interaction and flushing is pretty slow. That's something you'll figure out through testing.

Reply:
But how can we test this? It is mentioned in the Repo, that interactive problems aren't supported for automated testing :(